SubZero/src/model/filter/Filter.h

```diff
@@ -7,11 +7,42 @@
 
 #ifndef FILTER_H_
 #define FILTER_H_
+#include "DataDefinition.h"
+#include "RGBFilter.h"
+//#include "HSVFilter.h"
+
 
 class Filter {
+	/*
+	 * Filter is a virtual class that is further specified into specific
+	 * filters accessible by FilterFactory. In general filters will have
+	 * a filter() function that will filter input according to the
+	 * algorithm it contains and the parameters it holds.
+	 */
+
 public:
 	Filter();
 	virtual ~Filter();
+
+	// The filter function takes a pointer and applies the algorithm of the
+	// specific filter to the content of the pointer. The content of the
+	// pointer is altered and 0 is returned for successful filter, 1 for no
+	// action performed due to incorrect type, 2 for other errors.
+	virtual int filter(CamData * inputImg) {
+		return 1;
+	}
+	// Overload for FPGA filtering
+	virtual int filter(FPGAData * inputAttitude) {
+		return 1;
+	}
+
```

Comment:
Ah, good point. I intend to not modify the input in case of these "errors". What I am for seeing is if someone overloads a image filter with a fpga data in which case the filter cannot operate on the input. So in the virtual Filter class, I will return 1 to inform the FilterManager that the no action was performed by the filter. Is this a clear explanation? Let me know!